sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java

```diff
@@ -568,6 +568,29 @@ public Duration getAllowedTimestampSkew() {
   public @interface ProcessElement {}
 
   /**
+   * <b><i>Experimental - no backwards compatibility guarantees. The exact name or usage of this
+   * feature may change.</i></b>
+   *
+   * <p>Annotation that may be added to a {@link ProcessElement} or {@link OnTimer} method to
+   * indicate that the observable contents of the input {@link PCollection} or mutable state must be
+   * stable upon retries.
+   *
+   * <p>This is important for exactly-once semantics when writing to a storage medium outside of
```

Comment:
The language of this paragraph doesn't make it entirely obvious that this is mostly a concern for authors of Sinks.

Overall, I appreciate the depth of this documentation, but for a Beam user is feels intimidating and I'm not sure where/when I should use it.

Reply:
Added "runner" and "sink" in key phrases to try to clarify.